zerver/views/users.py

```diff
@@ -229,13 +229,22 @@ def regenerate_bot_api_key(request, user_profile, email):
     )
     return json_success(json_result)
 
+def add_outgoing_webhook_service(name, user_profile, base_url, interface, token):
+    # type: (Text, UserProfile, Text, int, Text) -> None
+    Service.objects.create(name=name,
+                           user_profile=user_profile,
+                           base_url=base_url,
+                           interface=interface,
+                           token=token)
+
 @has_request_variables
 def add_bot_backend(request, user_profile, full_name_raw=REQ("full_name"), short_name=REQ(),
-                    bot_type=REQ(validator=check_int, default=UserProfile.DEFAULT_BOT),
+                    bot_type=REQ(validator=check_int, default=UserProfile.DEFAULT_BOT), service_base_url=REQ(),
                     default_sending_stream_name=REQ('default_sending_stream', default=None),
                     default_events_register_stream_name=REQ('default_events_register_stream', default=None),
                     default_all_public_streams=REQ(validator=check_bool, default=None)):
-    # type: (HttpRequest, UserProfile, Text, Text, int, Optional[Text], Optional[Text], Optional[bool]) -> HttpResponse
+    # type: (HttpRequest, UserProfile, Text, Text, int, Text, Optional[Text], Optional[Text], Optional[bool]) -> HttpResponse
```

Comment:
Type for "service_base_url" could be 'Optional[Text]' instead of simple 'Text' ?